powerquery-docs/query-folding-examples.md

```diff
@@ -0,0 +1,290 @@
+
+After connecting to your database and navigating to the **fact_Sales** table, you select the **Keep bottom rows** transform found inside the Reduce rows group of the home tab as shown in the next image.
+
+![Keep bottom rows transform found inside the Reduce rows group of the home tab](media/query-folding-basics/keep-bottom-rows-ui.png)
+
+After selecting this transform, a new dialog will appear where you can enter the number of rows that you'd like to keep. For this case, you enter the value ten as shown in the image below and then click the OK button.
+
+![Entering the value ten inside the Keep bottom rows dialog](media/query-folding-basics/keep-bottom-rows-dialog.png)
+
+>[!TIP]
+>For this case, performing this operation yields the result of the last ten sales. In most scenarios it is recommended to provide a more explicit logic that defines which rows are considered last by applying a sort operation on the table.
+
+Next, you select the **Choose columns** transform found inside the *Manage columns* group from the Home tab which will let you select the columns that you want to keep from your table and remove the rest.
+
+![Selecting the choose columns transform found inside the Manage columns group from the Home tab](media/query-folding-basics/choose-columns-ui.png)
+
+Lastly, inside the **Choose columns** dialog, you select the columns *Sales Key*, *Customer Key*, *Invoice Date Key*, *Description*, and *Quantity* and click the OK button.
+
+![Selecting the columns Sales Key, Customer Key, Invoice Date Key, Description, and Quantity and click the OK button. inside the Choose columns dialog](media/query-folding-basics/choose-columns-dialog.png)
+
+This yields exactly the output that you were tasked with and below is the full M script for the query created:
+
+```
+let
+  Source = Sql.Database(ServerName, DatabaseName),
+  Navigation = Source{[Schema = "wwi", Item = "fact_Sale"]}[Data],
+  #"Kept bottom rows" = Table.LastN(Navigation, 10),
+  #"Choose columns" = Table.SelectColumns(#"Kept bottom rows", {"Sale Key", "Customer Key", "Invoice Date Key", "Description", "Quantity"})
+in
+  #"Choose columns""
+```
+
+### No query folding: Understanding the query evaluation
+
+Under “Applied steps”, you’ll notice that the step folding indicators for, Kept bottom rows and Choose columns are marked as steps that will be evaluated outside the data source or, in other words, by the Power Query engine.
+
+![Applied steps pane for the query with the step folding indicators showcasing that the Kept bottom rows and the Removed other columns steps are marked as steps that will be evaluated outside the data source](media/query-folding-basics/no-folding-steps.png)
+
+You can right click the last step of your query, the one named *Kept bottom rows*, and select the option that reads **Query plan** The goal of the Query plan is to showcase which transforms will be evaluated by the Power Query engine and which transforms could be offloaded to the data source.
```

Comment:
Doesn't Applied Steps already show this?

Reply:
Maybe we could rephrase this. 
AFAIK, The step folding indicators only showcases if the step folds or not. The query plan shows every single evaluation that will happen internally by Power Query. For example, if I try to combine multiple operations in a single step (imagine a Table.Buffer wrapped around a Table.Sort), the step will only show that it doesn't fold. The Query plan will tell me that it will do the sort locally and then the buffer. 

Maybe Jorge could help us here?

Reply:
Yes, that's correct. The query plan will work regardless on whether the steps follow the conventions generated by the UX or not, and it will only tell you the final outcome. Nuances about different operators in the same step are not exposed by the step indicators.

Reply:
@ehrenMSFT  should we add a note to be more explicit about the possible differences between relying solely on the step folding indicators vs checking the query plan?

Reply:
I think we should update the description of the "goal" of the Query Plan, since it does more than just show which transforms are folded vs. not. Its intent is to give you the nitty gritty details of how your query/step will actually be run.

Reply:
`You can right click the last step of your query, the one named *Kept bottom rows*, and select the option that reads **Query plan**. The goal of the query plan is to provide you with a better and more detailed view of your query's evaluation. You can learn more about this feature from the article on [Query plan](query-plan.md).
`
Would this work?

Reply:
Definitely better. I would say "The goal of the query plan is to provide you with a detailed view of how your query is actually run".

Reply:
Done.